### PR TITLE
add workaround fix viewer crash

### PR DESCRIPTION
**Pull request by j8xixo12**
Enhance #119.
Adding workaround to prevent viewer crash on MacOS by force QT using OpenGL API directly.

### CODE REVIEW

File: cpp/modmesh/view/wrap_view.cpp [hunk 1]
```diff
@@ -497,6 +497,12 @@ OneTimeInitializer<view_pymod_tag> & OneTimeInitializer<view_pymod_tag>::me()
 
 void initialize_view(pybind11::module & mod)
 {
+// TODO: There is some bug in QT RHI, this is a workaround to force QT use opengl directly
```
Comment (yungyuc) [inline cpp/modmesh/view/wrap_view.cpp:500]: Could you please add a reference (maybe URL) to where the QT RHI bug is described?

File: cpp/modmesh/CMakeLists.txt [hunk 2]
```diff
@@ -164,4 +164,12 @@ else () # MSVC
...
 endif () # MSVC
 
+if(QT3D_USE_RHI)
+    target_compile_options(modmesh_primary PRIVATE
+        ${COMMON_COMPILER_OPTIONS}
+        -DQT3D_USE_RHI
+    )
+endif() # QT3D_USE_RHI
+
+
```
Comment (yungyuc) [inline cpp/modmesh/CMakeLists.txt]: Please use a single blank line.

File: cpp/binary/viewer/CMakeLists.txt [hunk 3]
```diff
@@ -101,4 +101,4 @@ install(TARGETS viewer
     RUNTIME DESTINATION "${INSTALL_VIEWERDIR}"
     BUNDLE DESTINATION "${INSTALL_VIEWERDIR}"
     LIBRARY DESTINATION "${INSTALL_VIEWERDIR}"
-)
\ No newline at end of file
+)
```
Comment (yungyuc) [inline cpp/binary/viewer/CMakeLists.txt]: Please do not make this change that is only unnecessary formatting.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: cpp/modmesh/view/wrap_view.cpp (hunk 1 above)
Comment (j8xixo12) [reply cpp/modmesh/view/wrap_view.cpp:500]: fix in https://github.com/solvcon/modmesh/pull/195/commits/1727761ea359dd810c7be649c5128ef8e9310c1a

File: cpp/modmesh/CMakeLists.txt (hunk 2 above)
Comment (j8xixo12) [reply cpp/modmesh/CMakeLists.txt]: ditto

File: cpp/binary/viewer/CMakeLists.txt (hunk 3 above)
Comment (j8xixo12) [reply cpp/binary/viewer/CMakeLists.txt]: ditto

File: cpp/modmesh/view/wrap_view.cpp (hunk 1 above)
Comment (yungyuc) [reply cpp/modmesh/view/wrap_view.cpp:500]: Thanks

**Review by yungyuc — APPROVED** [review]
